crates/chia-consensus/src/multiprocess_validation.rs

```diff
@@ -0,0 +1,467 @@
+                    (
+                        pk,
+                        make_aggsig_final_message(*condition, msg.as_slice(), &spend, constants),
+                    )
+                })
+            })
+            .collect::<Vec<_>>()
+    });
+    let unsafe_items = npcresult
+        .agg_sig_unsafe
+        .iter()
+        .map(|(pk, msg)| (pk, msg.as_slice().to_vec()));
+    let iter = iter.chain(unsafe_items);
+    // Verify aggregated signature
+    let (result, added) = cache
+        .lock()
+        .unwrap()
+        .aggregate_verify(iter, &spend_bundle.aggregated_signature);
+    if !result {
+        return Err(ErrorCode::InvalidSpendBundle);
+    }
+    Ok((npcresult, added, start_time.elapsed()))
+}
+
+pub fn get_flags_for_height_and_constants(height: u32, constants: &ConsensusConstants) -> u32 {
+    let mut flags: u32 = 0;
+
+    if height >= constants.soft_fork4_height {
+        flags |= ENABLE_MESSAGE_CONDITIONS;
+    }
+
+    if height >= constants.soft_fork5_height {
+        flags |= DISALLOW_INFINITY_G1;
+    }
+
+    if height >= constants.hard_fork_height {
+        //  the hard-fork initiated with 2.0. To activate June 2024
+        //  * costs are ascribed to some unknown condition codes, to allow for
+        // soft-forking in new conditions with cost
+        //  * a new condition, SOFTFORK, is added which takes a first parameter to
+        //    specify its cost. This allows soft-forks similar to the softfork
+        //    operator
+        //  * BLS operators introduced in the soft-fork (behind the softfork
+        //    guard) are made available outside of the guard.
+        //  * division with negative numbers are allowed, and round toward
+        //    negative infinity
+        //  * AGG_SIG_* conditions are allowed to have unknown additional
+        //    arguments
+        //  * Allow the block generator to be serialized with the improved clvm
+        //   serialization format (with back-references)
+        flags = flags
+            | ENABLE_SOFTFORK_CONDITION
+            | ENABLE_BLS_OPS_OUTSIDE_GUARD
+            | ENABLE_FIXED_DIV
+            | AGG_SIG_ARGS
+            | ALLOW_BACKREFS;
+    }
+    flags
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::consensus_constants::TEST_CONSTANTS;
+    use crate::gen::condition_tools::u64_to_bytes;
+    use chia_bls::{sign, SecretKey, Signature};
+    use chia_protocol::{Bytes, Bytes32};
+    use chia_protocol::{Coin, CoinSpend, Program};
+    use clvm_utils::tree_hash_atom;
+    use hex::FromHex;
+    use hex_literal::hex;
+
+    #[test]
+    fn test_validate_no_pks() {
+        let test_coin = Coin::new(
+            Bytes32::new(hex!(
+                "4444444444444444444444444444444444444444444444444444444444444444"
+            )),
+            Bytes32::new(hex!(
+                "3333333333333333333333333333333333333333333333333333333333333333"
+            )),
+            1,
+        );
```

Comment:
I think you can simplify this to:
```suggestion
        let test_coin = Coin::new(
            hex!("4444444444444444444444444444444444444444444444444444444444444444").into(),
            hex!("3333333333333333333333333333333333333333333333333333333333333333").into(),
            1,
        );
```

Reply:
Done